src/init.cpp

```diff
@@ -1590,6 +1590,8 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
     if (fPruneMode) {
         LogPrintf("Unsetting NODE_NETWORK on prune mode\n");
         nLocalServices = ServiceFlags(nLocalServices & ~NODE_NETWORK);
+        // always set NODE_NETWORK_LIMITED (BIP159) in pruned mode
+        nLocalServices = ServiceFlags(nLocalServices | NODE_NETWORK_LIMITED);
```

Comment:
Hmm, I thought we were just going to alays set this? Why bother *only* setting it when pruned - a node that is not pruned is also capable of serving the last 288 blocks, which is the way I read the definition of NODE_NETWORK_LIMITED.

Reply:
I agree. Came to complain about the same thing.

NODE_NETWORK implies NODE_NETWORK_LIMITED, so _LIMITED should always be set.
Otherwise, we're introducing negative service flags, which is a bad idea imo.

Reply:
Okay. Agree... will change it.